src/main/java/heronarts/lx/Tempo.java

```diff
@@ -191,16 +194,38 @@ public Tempo(LX lx) {
     addParameter("trigger", this.trigger);
     addParameter("enabled", this.enabled);
     addModulator("nudge", this.nudge);
+
+    // set initial minBpm, maxBpm
+    minBpm = DEFAULT_MIN_BPM;
+    maxBpm = DEFAULT_MAX_BPM;
   }
 
   private static final String PATH_BEAT = "beat";
   private static final String PATH_SET_BPM = "setBPM";
+  
+  public boolean setBpmRange(double min, double max) {
+    if (min <= 0.0 || min >= max) {
+      // do not set to invalid range of BPMs
+      return false;
+    }
+    
+    minBpm = min;
+    maxBpm = max;
+    return true;
+  }
+
+  public boolean isValidBpm(double bpm) {
+    return bpm >= minBpm && bpm <= maxBpm;
```

Comment:
LX coding style nitpick - I'm obnoxious about always using `this.minBpm` and `this.maxBpm` to refer to member variables

Reply:
i actually am totally in the "write out this. everytime camp". glad to see someone else is too

Reply:
we don't always in TE